### PR TITLE
Add CodeQL Analysis

### PR DESCRIPTION
**Pull request by devlead**

### CODE REVIEW

**Review by nils-a — APPROVED** [review]
LGTM!

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ develop, main, release/1.0.0 ]
+  pull_request:
+    branches: [ develop, release/1.0.0 ]
+  schedule:
+    - cron: '41 21 * * 2'
```
Comment (augustoproiete) [inline .github/workflows/codeql-analysis.yml:9]: At 21:41 on Tuesday? That's random 😄

**Review by augustoproiete — APPROVED** [review]
LGTM

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (devlead) [reply .github/workflows/codeql-analysis.yml:9]: Yeah, their template picks time at random 😉

**Review by mholo65 — APPROVED** [review]

**Pull request merged**